lib/autocomplete-manager.coffee

```diff
@@ -272,18 +186,166 @@ class AutocompleteManager
     @editor.insertText(match.word)
     @editor.setSelectedBufferRanges(newSelectedBufferRanges)
 
-  # Private: Sets the current buffer, starts listening to change events and delegates
-  # them to #onChanged()
+  # Public: Finds and returns the content before the current cursor position
+  #
+  # selection - The {Selection} for the current cursor position
+  #
+  # Returns {String} with the prefix of the {Selection}
+  prefixOfSelection: (selection) ->
+    selectionRange = selection.getBufferRange()
+    lineRange = [[selectionRange.start.row, 0], [selectionRange.end.row, @editor.lineTextForBufferRow(selectionRange.end.row).length]]
+    prefix = ''
+    wordRegex = /\b\w*[a-zA-Z_-]+\w*\b/g
```

Comment:
It certainly feels like we should find some way to leverage the grammars to do the heavy lifting to determine prefix by default. The current method is clunky and causing too many bugs.